File: app/src/main/java/com/example/benchmarks/BenchmarksApplication.java
package com.example.benchmarks;

import android.app.Application;

import com.example.benchmarks.models.AppComponent;
import com.example.benchmarks.models.DaggerAppComponent;

public class BenchmarksApplication extends Application {

    private static AppComponent appComponent;

    @Override
    public void onCreate() {
        super.onCreate();
        appComponent = DaggerAppComponent.create();
    }

    public static void setAppComponent(AppComponent component){
        appComponent = component;
    }

    public static AppComponent getAppComponent() {
        return appComponent;
    }

}

File: app/src/main/java/com/example/benchmarks/BenchmarksApplication.kt
package com.example.benchmarks

import android.app.Application
import com.example.benchmarks.models.AppComponent
import com.example.benchmarks.models.DaggerAppComponent

class BenchmarksApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        appComponent = DaggerAppComponent.create()
    }

    companion object {
        private var appComponent: AppComponent? = null
        @JvmStatic
        fun setAppComponent(component: AppComponent?) {
            appComponent = component
        }

        @JvmStatic
       fun getAppComponent(): AppComponent? {
            return appComponent
        }
    }
}